### PR TITLE
#2 concrete strength prediction with svm

### PR DESCRIPTION
**Pull request by saidsabri010**
i read the dataset and train_test_split it,and created the svm class that get the score (SVR) and store the result in 
CSV file,for test,i checked if the output is an instance of pandas dataframe

### CODE REVIEW

File: src/svm.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                  'gamma': [1, 0.1, 0.01, 0.001, 0.0001],
+                  'kernel': ['rbf']}
+    grid = GridSearchCV(SVR(), param_grid, refit=True, verbose=3)
+    grid.fit(X_train, y_train)
+    y_pred = grid.predict(X_test)
+    y_pred = sc_y.inverse_transform(y_pred)
+    score = grid.score(X_test, y_test)
+
+    # save the score in a csv file
+    fields = ['Score']
```
Comment (kadamQ) [inline src/svm.py:41]: - The best score can change, so you should try to save the scores dynamically, not giving concrete values to the variables.
- Try to save the [best score](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html) with the [best params ](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html)too. 
- For this use `best_score_` and `best_params_`.

**Review by kadamQ — CHANGES_REQUESTED** [review]

File: src/svm.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+              }
+
+
+class SupportVectorMachine:
+    """
+    class for svm
+    """
+
+    def __init__(self):
+        self.grid = GridSearchCV(SVR(), param_grid, refit=True, verbose=3)
```
Comment (Benczus) [inline src/svm.py:26]: Param Grid and Model can be taken as a parameter

File: src/svm.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        data_x = preprocessing.scale(data_x)
+        data_y = preprocessing.scale(data_y)
+        scalar.fit_transform(data_x)
+        scalar.fit_transform(data_y)
+        self.grid.fit(train_x, y_train)
+        self.grid.score(test_x, y_test)
+        # grid.best_estimator_
+        return self.grid.cv_results_
+
+    def save(self):
```
Comment (Benczus) [inline src/svm.py:50]: visualization is missing

**Review by Benczus — CHANGES_REQUESTED** [review]
Visualization of the regressor is important part of machine learning.
Please visualize the original data points with a scatter plot by taking a column of the dataset for the x and the target for the y axis.
Then use scatter plot to plot the same column with the predicted values as reference.
Since you have used standard scaler, please don't forget to use .inverse_transform(y_predict) before plotting
